lib/ramble/docs/tutorials/using_zips_and_matrices.rst

```diff
@@ -0,0 +1,354 @@
+        mpi_command: mpirun -n {n_ranks}
+      applications:
+        wrfv4:
+          workloads:
+            CONUS_12km:
+              experiments:
+                scaling_{n_nodes}:
+                  variables:
+                    n_nodes: [1, 2, 4]
+      spack:
+        concretized: true
+        packages:
+          gcc9:
+            spack_spec: gcc@9.3.0
+          intel-mpi:
+            spack_spec: intel-mpi@2018.4.274
+            compiler: gcc9
+          wrfv4:
+            spack_spec: wrf@4.2 build_type=dm+sm compile_type=em_real nesting=basic ~chem
+              ~pnetcdf
+            compiler: gcc9
+        environments:
+          wrfv4:
+            packages:
+            - intel-mpi
+            - wrfv4
+
+The above configuration will execute 3 experiments, comprising a basic scaling
+study on three different sets of nodes.
+
+We will now expand this to perform more experiments using the zip
+(:ref:`ramble-explicit-zips`) and matrix (:ref:`ramble-matrix-logic`)
+functionality in Ramble.
+
+Construct Platforms Zip
+-----------------------
+
+For the purposes of this tutorial, you will construct a zip representing
+multiple platforms. The platforms will differ by their value of the
+``processes_per_node`` variable.
+
+Zips are explicit groupings of variables, that are combined into a larger
+variable set. A zip is defined by a list of variable definitions, where each
+individual variable is a list / vector variable and all variables are the same
+length. As an example, imagine we had the following variable / zip definitions:
+
+.. code-block:: YAML
+
+    variables:
+      platform: ['platform1', 'platform2']
+      processes_per_node: ['2', '4']
+    zips:
+      platform_config:
+      - platform
+      - processes_per_node
+
+The result of this is that ``platform_config`` would be a list of length 2. The
+first index would contain ``(platform1, 2)`` and the second index would contain
+``(platform2, 4)``. Using this, we can group an arbitrary number of variables
+into a single name.
+
+For the purposes of this tutorial, we'll assume your system has 4 total cores,
+allowing us to use the platform definitions from the above YAML.
+
+Edit your workspace configuration file to include the ``platform_config`` from
+the above section. The result should look like the following:
+
+.. code-block:: YAML
+
+    ramble:
+      variables:
+        processes_per_node: 4
+        n_ranks: '{processes_per_node}*{n_nodes}'
+        batch_submit: '{execute_experiment}'
+        mpi_command: mpirun -n {n_ranks}
+        platform: ['platform1', 'platform2']
+        processes_per_node: ['2', '4']
+      zips:
+        platform_config:
+        - platform
+        - processes_per_node
+      applications:
+        wrfv4:
+          workloads:
+            CONUS_12km:
+              experiments:
+                scaling_{n_nodes}:
+                  variables:
+                    n_nodes: [1, 2, 4]
```

Comment:
Does this work? Don't you need to matrix it in to avoid having vecs of len 2 and 3?

Reply:
I kind of wanted to lead users into the error case without them expecting it. That way they are forced to see how it breaks and then fix it.

Reply:
OK sounds good!